platform/http_session_manager.mm

```diff
@@ -111,7 +111,7 @@ - (void)URLSession:(NSURLSession *)session
   if ([taskInfo.delegate
        respondsToSelector:@selector(URLSession:task:willPerformHTTPRedirection:newRequest:completionHandler:)])
   {
-    dispatch_async(dispatch_get_main_queue(), ^{
+    dispatch_async(dispatch_get_current_queue(), ^{
```

Comment:
For some reason tests on MacOS failed with `dispatch_get_main_queue()` function. Replaced with `dispatch_get_current_queue()`. Looks like main queue != current queue. It doesn't break iOS.

But `dispatch_get_current_queue()` function is deprecated. What is propper fix here?

Reply:
Because of my change tests are failing at [assertion](https://github.com/organicmaps/organicmaps/blob/master/platform/http_request.cpp#L185):

```
ASSERT_EQUAL(id, threads::GetCurrentThreadID(), ("OnWrite called from different threads"));
```

How to fix it on MacOS? Remove assert?

Reply:
HTTP code is thread-safe only if it's called on the same thread. Instead of removing the assert, the callbacks logic should be checked, why different thread is used in different `OnWrite` calls.

Reply:
Looks like there is no such thing as main thread in our test apps.
If you go to [downloader_test.cpp](https://github.com/organicmaps/organicmaps/blob/master/platform/platform_tests/downloader_tests/downloader_test.cpp) you can see calls to `QCoreApplication::exec()`.

I think, main queue is created only after `QCoreApplication::exec()` call. But it doesn't work for `osm_auth_tests.cpp`. I'm getting error:

```
QApplication::exec: Please instantiate the QApplication object first
```

Reply:
This is how good stack trace looks like when running `platform_tests`:
![good-stack-trace](https://github.com/organicmaps/organicmaps/assets/720808/48bdf08b-118e-4866-b5b1-08814aa44da8)

`dispatch_async(dispatch_get_main_queue(), ...)` Realy dispatches code in main queue. Need to do simillar for `osm_auth_tests`.

Reply:
@biodranik  I found a workaround for MacOS: see commit [#6c8171f8](https://github.com/organicmaps/organicmaps/pull/7333/commits/6c8171f83786773ebf19905dde2d4e468435ec25).

For MacOS each tests should be wrapped in lambda and executed in a separate thread while main thread is used for main queue.

Do you have any ideas how to improve this solution?

Reply:
@biodranik I changed `HttpSessionManager` behaviour in commit [#117527f](https://github.com/organicmaps/organicmaps/pull/7333/commits/117527f2c7ef4a7335cc8a3c87227a63ba29208f)
Instead of using main queue for all callbacks it now uses custom queue.

Tests `oam_auth_tests` and `platform_tests` are not failing.

Reply:
👆 Update: that solution doesn't actually work.

We can't replace main queue with custom queue because there is no guaranty that the same thread will be used. When we post task to a queue with `dispatch_async(...)` it could be handled from different threads. See https://stackoverflow.com/q/52389851

So we should either keep main queue and use some workarounds for unit-tests or get rid of queues and use threads in `HttpSessionManager` class.